ui/src/VoxettaUserForm.js

```diff
@@ -187,69 +201,53 @@ export class VoxettaUserForm extends LitElement {
         this.userAge = this.shadowRoot.getElementById('user-age').value;
         this.deviceType = this.shadowRoot.getElementById('device-type').value;
 
-        if (this.ensureFormCompleteness()) {
-            const userInfo = {
-                userId: this.userId, 
-                gender: this.gender,
-                userAge: this.userAge,
-                deviceType: this.deviceType
-            };
-            this.cookieService.makeUserInfoCookie(userInfo);
-            this.handleExitForm(); 
-        } 
+        const userInfo = {
+            userId: this.userId, 
+            gender: this.gender,
+            userAge: this.userAge,
+            deviceType: this.deviceType
+        };
+
+        this.handleFormSubmission(userInfo);
+        this.handleExitForm(); 
     }
 
     /**
-     * Ensures the validity of a submitted form in regards to completeness. 
-     * @returns {Boolean} Denotes if the submitted form was fully completed.
+     * Determines if each input in the user form is valid and alters Save Button appropriately. 
      */
-    ensureFormCompleteness() {
-        const idComplete = 
-            this.ensureInputCompleteness('user-id', 'Please enter a valid user id.');
-        const genderComplete =
-            this.ensureInputCompleteness('gender-list', 'Please select a gender.');
-        const ageComplete =
-            this.ensureInputCompleteness('user-age', 
-                'Please enter a valid age between 0 and 120.');
-        const deviceComplete =
-            this.ensureInputCompleteness('device-type', 'Please enter a device name.');
+    formIsValid() {
+        const userIdValidity = this.shadowRoot.getElementById('user-id').checkValidity();
```

Comment:
can maybe squish this a bit

```javascript
const form = this.shadowRoot.getElementById('my-cool-form');
const valid = form.elements.every((element) => {
  return element.checkValidity();
});
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every
https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/elements

Reply:
I unfortunately don't think the .elements function is applicable because I am not using the components that this function captures, I also tried using .childNodes but this led to an issue with the select div (necessary style workaround--currently there is know bug with `<mwc-select>` width)